tools/deno

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env -S deno run --unstable --allow-read --allow-write --allow-env --no-check
+
+/* eslint no-console: "off" */
+/* eslint no-process-exit: "off" */
+
+import { existsSync } from "https://deno.land/std@0.90.0/fs/exists.ts";
+
+import { DefaultTestFileLoader } from "../out/common/framework/file_loader.js";
+import { Logger } from "../out/common/framework/logging/logger.js";
+import { parseQuery } from "../out/common/framework/query/parseQuery.js";
+import { assert, unreachable } from "../out/common/framework/util/util.js";
+
+function usage(rc) {
```

Comment:
Would it be possible to move some of this code into TypeScript so it can be shared with the code in `src/common/runtime/cmdline.ts`, e.g. `runtime/deno.ts`? Since you're using `npm run standalone` it will transpile these to .js as well.

Happy to land this PR without this, but would be great to work on it as a follow-up.

Reply:
There are two possible options, im open to either:

1. Deno can run TS out of the box. I can convert the `./tools/deno` file to `./tools/deno.ts`, but this would still have to import from the `./out` dir, and would thus not get typings for the imports. This is because I can't import the TS files directly from source, because Deno does not understand them, because deno uses fully specified import specifiers, like the web, which TSC does not support.
2. Add a `src/common/runtime/deno.ts`, add `// @ts-ignore` to the fully qualified http import (because tsc does not understand it), and add inline type definitions for the parts of the `Deno` global we use.

I think 2 is far easier, but maybe also overcomplicated. A possibly better third approach might be to just factor all of the common code into a single ./runtime/cmdline.ts module, which exports a `init` function that takes an object containing all of the system APIs (like `readFile`, `writeFile`, `env`, `args`, etc). We can then have a simple `./tools/node` and `./tools/deno` which both import this init function and pass their respective APIs to it.

Reply:
3 sounds great to me. (My only other suggestion would be 2 but using a bunch of `any` instead of actually writing inline type definitions, it would be annoying.)

I don't fully understand the issue with 1. Except for a few utilities that only run in Node, all of our import specifiers are relative (and web-compatible as that's the main place we run them). We can have TSC ignore the whole file if that would help?
Anyway, it's okay, I don't need to understand the issue :)